core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -56,21 +65,30 @@ public AtomicFileWriter(File f) throws IOException {
     }
 
     /**
-     * @param encoding
-     *      File encoding to write. If null, platform default encoding is chosen.
+     * @param encoding File encoding to write. If null, platform default encoding is chosen.
+     *
+     * @deprecated Use {@link #AtomicFileWriter(File, Charset)}
      */
     public AtomicFileWriter(File f, String encoding) throws IOException {
-        File dir = f.getParentFile();
+        this(f, Charset.forName(encoding));
+    }
+
+        /**
+         * @param charset File charset to write. If null, platform default encoding is chosen.
+         */
+    public AtomicFileWriter(File f, Charset charset) throws IOException {
+        destFile = f.toPath();
```

Comment:
f.toPath may throw an [`InvalidPathException`](https://docs.oracle.com/javase/8/docs/api/java/nio/file/InvalidPathException.html) which is not an `IOException`.

Reply:
Yeah, but it's a `IllegalArgumentException` which is a runtime anyway. So, well. Can add javadoc, or wrap this all to preserve previous behaviour on previous code paths, and propagate on new ones.